library/tract-renderer/src/main/java/org/cru/godtools/tract/activity/TractActivity.java

```diff
@@ -320,8 +342,26 @@ private boolean isDeepLinkValid(@Nullable final Uri data) {
     }
 
     @NonNull
-    private Locale getLanguageFromDeepLink(@NonNull final Uri data) {
-        return LocaleCompat.forLanguageTag(data.getPathSegments().get(0));
+    private Locale[] processDeepLinkLanguages(@NonNull final Uri data) {
+        // extract raw locales from the uri
+        final Locale uriLocale = LocaleCompat.forLanguageTag(data.getPathSegments().get(0));
+        final List<Locale> rawPrimaryLanguages = Stream.of(data.getQueryParameters(PARAM_PRIMARY_LANGUAGE))
+                .flatMap(lang -> Stream.of(TextUtils.split(lang, ",")))
+                .map(String::trim)
+                .filterNot(TextUtils::isEmpty)
+                .map(LocaleCompat::forLanguageTag)
+                .toList();
+
+        // process the primary languages specified in the uri
+        if (!rawPrimaryLanguages.isEmpty()) {
```

Comment:
Preference: This may be a swift/ruby convention, but I now prefer in a case like this to flip the logic check here and in the case that `rawPrimaryLanguages` is empty, return the uriLocal. It indents less code is more readable in my opinion